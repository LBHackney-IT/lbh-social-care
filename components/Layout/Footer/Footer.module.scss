@import 'lbh-frontend/lbh/base';

.footer {
  background: lbh-colour('lbh-text');
  color: lbh-colour('lbh-white');
  font-size: 0.9rem;
  padding: 30px 0;

  a,
  a:visited,
  a:link {
    color: lbh-colour('lbh-white');
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
    &:focus {
      color: lbh-colour('lbh-text');
    }
  }
}

.links {
  margin-bottom: 5px;
  font-size: 1rem;

  a {
    display: block;
    @include govuk-media-query($from: tablet) {
      display: inline-block;
      margin-top: 0;
      margin-right: 20px;
    }
  }
}

.meta {
  color: lbh-colour('lbh-grey-3');
}
