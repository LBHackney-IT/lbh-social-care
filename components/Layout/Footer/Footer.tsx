import s from './Footer.module.scss';

const Footer = (): React.ReactElement => (
  <footer className={`${s.footer}`} role="contentinfo">
    <div className="govuk-width-container">
      <nav className={s.links}>
        <a href="https://sites.google.com/hackney.gov.uk/moderntoolsforsocialcare/home">
          Roadmap
        </a>
        <a href="https://docs.google.com/forms/d/e/1FAIpQLScILbPD1ioKHzp1D3HN4_DKaxV2tpWLMu8upSSqNgSPCo85cg/viewform">
          Give feedback
        </a>
      </nav>

      <div className={s.meta}>Built and maintained by HackIT.</div>
    </div>
  </footer>
);

export default Footer;
