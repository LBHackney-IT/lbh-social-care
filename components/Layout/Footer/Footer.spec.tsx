import { render } from '@testing-library/react';

import Footer from './Footer';

jest.mock('./OGL', () => () => 'MockIcon');

describe('Footer', () => {
  it('should render properly', async () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
