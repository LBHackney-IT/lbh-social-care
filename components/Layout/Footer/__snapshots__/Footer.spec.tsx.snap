// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render properly 1`] = `
<DocumentFragment>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          MockIcon
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
